src/elexmodel/models/BaseElectionModel.py

```diff
@@ -69,21 +69,25 @@ def get_unit_predictions(self, reporting_units, nonreporting_units, estimand):
         nonreporting_units_features = self.featurizer.featurize_heldout_data(nonreporting_units)
 
         weights = reporting_units[f"last_election_results_{estimand}"]
-        reporting_units_residuals = reporting_units[f"residuals_{estimand}"]
+        reporting_units_pp_changes = reporting_units[f"pp_change_{estimand}"]
 
-        self.fit_model(self.qr, reporting_units_features, reporting_units_residuals, 0.5, weights, True)
+        self.fit_model(self.qr, reporting_units_features, reporting_units_pp_changes, 0.5, weights, True)
         self.features_to_coefficients = dict(zip(self.featurizer.complete_features, self.qr.coefficients))
 
         preds = self.qr.predict(nonreporting_units_features)
 
-        # multiply by total voters to get unnormalized residuals
-        preds = preds * nonreporting_units[f"last_election_results_{estimand}"]
+        # Must run turnout first!
+        # We now get the final RAW number of predicted votes for an estimand by adding the
+        # preds (which are expected percentage-point change) to the baseline's estimand
+        # share, and multiplying the result by the current year's predicted turnout
+        if estimand == "turnout":
+            preds = (preds + nonreporting_units["baseline_turnout"]) * nonreporting_units["total_people"]
+            self.raw_turnout_preds = preds.copy()
 
-        # add in last election results to go from residual to number of votes in this election
-        # max with results so that predictions are always at least ars large as actual number of votes
-        preds = np.maximum(
-            preds + nonreporting_units[f"last_election_results_{estimand}"], nonreporting_units[f"results_{estimand}"]
-        )
+        else:
+            preds = (preds + nonreporting_units[f"baseline_{estimand}"]) * self.raw_turnout_preds
+
+        preds = np.maximum(preds, nonreporting_units[f"results_{estimand}"])
```

Comment:
I'm confused why this maximum is still necessary?

Reply:
we are unwound to raw votes at this point, so still taking max to make sure we're not under the actual results. Dem/rep raw votes are taken as a share of predicted turnout, so stll possible it comes out < alread-reported actual dem/rep votes